core/internal/featurechecker/features.go

```diff
@@ -0,0 +1,94 @@
+package featurechecker
+
+import (
+	"context"
+	"sync"
+
+	"github.com/Khan/genqlient/graphql"
+	"github.com/wandb/wandb/core/internal/gql"
+	"github.com/wandb/wandb/core/internal/observability"
+	spb "github.com/wandb/wandb/core/pkg/service_go_proto"
+)
+
+// ServerFeaturesCache is responsible for providing the capabilities of a server
+// Features and capabilities are retrieved from the server via a GraphQL query
+type ServerFeaturesCache struct {
+	Features map[spb.ServerFeature]Feature
+
+	ctx           context.Context
+	graphqlClient graphql.Client
+	logger        *observability.CoreLogger
+	once          sync.Once
+}
+
+type Feature struct {
+	DefaultValue string
+	Description  string
+	Enabled      bool
+	Name         string
+}
+
+func NewServerFeaturesCache(
+	ctx context.Context,
+	graphqlClient graphql.Client,
+	logger *observability.CoreLogger,
+) *ServerFeaturesCache {
+	return &ServerFeaturesCache{
+		ctx:           ctx,
+		graphqlClient: graphqlClient,
+		logger:        logger,
+		once:          sync.Once{},
+	}
+}
+
+func (sf *ServerFeaturesCache) loadFeatures() (map[spb.ServerFeature]Feature, error) {
+	features := make(map[spb.ServerFeature]Feature)
+
+	if sf.graphqlClient == nil {
+		sf.logger.Warn("GraphQL client is nil, skipping feature loading")
+		return features, nil
+	}
+
+	// Query the server for the features provided by the server
+	resp, err := gql.ServerFeaturesQuery(sf.ctx, sf.graphqlClient, gql.RampIDTypeUsername)
+	if err != nil {
+		sf.logger.Error(
+			"Failed to load features, feature will default to disabled",
+			"error",
+			err,
+		)
+		return features, err
```

Comment:
to cover this one you need to extend the mockgql to return errors, i'm not sure if it does it now

Reply:
Added a stub with errors to allow mockgql to throw errors.